docker_cli/_exceptions.py

```diff
@@ -0,0 +1,38 @@
+class CommandNotFoundError(Exception):
+    def __init__(self, command_name: str):
+        self.command_name = command_name
+        self.message = (f"Command '{self.command_name}' was run, but is not "
+                        "present on the Docker image.")
+
+    def __str__(self):
+        return self.message
+
+
+class DockerBuildError(Exception):
+    """Raised when a docker image fails to build"""
+
+    def __init__(self, image_tag: str, dockerfile: str, message: str = ""):
```

Comment:
Looks like this is intended to be a set of overloads? That is, either `image_tag` & `dockerfile` OR `message` should be specified, but not all 3.

```suggestion
    @typing.overload
    def __init__(self, image_tag: str, dockerfile: str):
        ...
        
    @typing.overload
    def __init__(self, message: str):
        ...
    
    def __init__(self, image_tag = None, dockerfile = None, message = None):
```

Reply:
Let's add docstrings explaining these parameters as well.

Reply:
And let's add a check that the user provided either `message` or `image_tag` + `dockerfile`, but not all 3.